joint_state_publisher/joint_state_publisher.py

```diff
@@ -483,7 +482,19 @@ def main(input_args=None):
     with open(parsed_args.urdf_file, 'r') as infp:
         urdf = infp.read()
 
-    node = rclpy.create_node('joint_state_publisher')
+    initial_parameters=[
+        rclpy.parameter.Parameter('num_rows', rclpy.parameter.Parameter.Type.INTEGER, 0),
+        rclpy.parameter.Parameter('publish_default_efforts', rclpy.parameter.Parameter.Type.BOOL, False),
+        rclpy.parameter.Parameter('publish_default_positions', rclpy.parameter.Parameter.Type.BOOL, True),
+        rclpy.parameter.Parameter('publish_default_velocities', rclpy.parameter.Parameter.Type.BOOL, False),
+        rclpy.parameter.Parameter('rate', rclpy.parameter.Parameter.Type.INTEGER, 10),  # 10hz
+        rclpy.parameter.Parameter('source_list', rclpy.parameter.Parameter.Type.STRING_ARRAY, []),
+        rclpy.parameter.Parameter('use_gui', rclpy.parameter.Parameter.Type.BOOL, False),
+        rclpy.parameter.Parameter('use_mimic_tags', rclpy.parameter.Parameter.Type.BOOL, True),
+        rclpy.parameter.Parameter('use_smallest_joint_limits', rclpy.parameter.Parameter.Type.BOOL, True),
```

Comment:
This node needs `node.declare_parameter()` calls corresponding to each of these parameters.